test/unit/partials/plugin/filter.js

```diff
@@ -0,0 +1,24 @@
+class FilterPlugin {
+  constructor(config) {
+    this._config = config;
+  }
+
+  filter_metadata(pkg) {
+    return new Promise((resolve) => {
```

Comment:
Didn't you forget 'reject' there ?

Reply:
Meh yes. I guess it threw inside the promise anyway so it still worked. You can fix in your patch as well.